packages/components/MIGRATING.md

```diff
@@ -66,6 +69,105 @@ Changes:
+
+        -   _optional_ - The `Dialog.Footer` can be passed in as a child of `Dialog.Content` to add a styled footer to the dialog.
+
+        -   _optional_ - The `Dialog.SideContent` can be passed in as a child of `Dialog.Content` to add a space on the left side of the dialog. Custom components can then be passed in as as children of `Dialog.SideContent`
+
+-   The styling / layout is now controlled on the subcomponent `Dialog.Content`
+
+-   The subcomponent `Dialog.Content` is used to display the dialog container.
+    It provides no styling by default and can be used to wrap a custom content.
+
+    -   `roundedCorners` is now called `rounded` and is a boolean.
+
+    -   The `width` and `maxHeight` props were removed. The Dialog container adjusts to the content inside. You can use a custom component inside if neccessary. you can use `minWidth`, `maxWidth` and `minHeight` to override the default values.
+
+    -   The `strategy` and `role` props were removed to simplify the API.
+
+    -   The `enablePortal` and `verticalAlignment` props were removed to standardize the dialog.
+
+        The dialog now uses a portal by default and is placed centered
+
+-   The subcomponents `Dialog.Header`, `Dialog.Body`, `Dialog.Footer` and `Dialog.SideContent` can be used to add defaut styling to the content inside of `Dialog.Content`.
+
+#### Old
+
+```tsx
+const [isOpen, setIsOpen] = useState(false);
+
+return (
+    <Button icon={<IconIcon/>} ref={triggerRef} onClick={() => setIsOpen(!isOpen)}>
+        Dialog Trigger
+    </Button>
+
+    <Dialog {...args} anchor={triggerRef} open={isOpen} handleClose={() => setIsOpen(false)}>
+        <DialogBody padding="spacious">
+            <p>
+                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad illum impedit iure numquam praesentium vel.
+                Distinctio perferendis, suscipit! Dolor doloremque et ex, modi nobis officiis perspiciatis quis tempora
+                temporibus voluptates?
+            </p>
+        </DialogBody>
+        <DialogFooter
+            actionButtons={[
+                {
+                    children: 'Close',
+                    onClick: () => {},
+                },
+            ]}
+        ></DialogFooter>
+    </Dialog>
+)
+```
+
+#### New
+
+```tsx
+<Dialog.Root>
+    <Dialog.Trigger>
+        <Button>Dialog Trigger</Button>
+    </Dialog.Trigger>
+    <Dialog.Content {...args}>
+        <Dialog.SideContent>
+            <div className="tw-bg-box-positive-strong tw-h-full tw-w-full"></div>
+        </Dialog.SideContent>
+        <Dialog.Header>Header</Dialog.Header>
+        <Dialog.Body padding="spacious">
+            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad illum impedit iure numquam praesentium vel.
+            Distinctio perferendis, suscipit! Dolor doloremque et ex, modi nobis officiis perspiciatis quis tempora
+            temporibus voluptates?
+        </Dialog.Body>
+        <Dialog.Footer>
```

Comment:
Should we add that gap by default to the footer style to avoid extra div everytime?
I also forgot to add that comment on the flyout but would be nice to also update it there 😬

Reply:
no because it does not neccessarily have to have buttons in it. It can be anything

Reply:
Most typical use cases is having buttons (you can see it from the shape of the older Dialog/Modal)

Reply:
we can add it if you feel like it helps

Reply:
added